pubsub.go

```diff
@@ -621,6 +614,43 @@ func (p *PubSub) processLoop(ctx context.Context) {
 	}
 }
 
+func (p *PubSub) handlePendingPeers() {
+	select {
+	case <-p.newPeersSema:
+		defer func() {
+			p.newPeersSema <- struct{}{}
+		}()
+
+	default:
+		// contention, return and wait for the next notification without blocking the event loop
+		return
```

Comment:
This seems like it might be ok, although is it a problem if there's contention for a long time?

Reply:
i don't think so, at worst some peers wont be processed for a while; we can live with that.